File: dcp-solutions/2024-12-02/solution.c
```c
// Definition for a binary tree node.
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    struct TreeNode *next;
};

struct TreeNode* invertTree(struct TreeNode* root) {
    if (root == NULL)
        return NULL;

    // Swap the left and right child nodes
    struct TreeNode* temp = root->left;
    root->left = root->right;
    root->right = temp;

    // Recursively call the function for both child nodes
    invertTree(root->left);
    invertTree(root->right);

    return root;
}
```

File: dcp-solutions/2024-12-02/solution.py
```python
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def invertTree(root: TreeNode) -> TreeNode:
    if not root:
        return None
    
    # Swap the values of left and right nodes
    root.val, root.left.val, root.right.val = root.right.val, root.left.val, root.right.val
    
    # Recursively invert the left and right subtrees
    temp = root.left
    root.left = invertTree(root.right)
    root.right = temp
    
    return root
```